### DISCUSSION TITLE
How to mock HttpRequest.Query?

### DISCUSSION THREAD

**Discussion by lommez** [dotnet/aspnetcore#26725] [General] [answered]
Is possible to mock HttpRequest.Query property ?

When i try to executed my test i got the following error:

Could not load type 'Microsoft.AspNetCore.Http.Internal.QueryCollection' from assembly 'Microsoft.AspNetCore.Http, Version=3.1.6.0,

**Comment by Tratcher** [comment, marked as answer]
I don't know why you get that error, but you shouldn't need mocks for HttpRequest. All of the fields were designed to be set to make them easy to test. For Query it might be easier to set HttpRequest.QueryString and let it populate Query for you.

**Reply by lommez** [reply]
Taking a look at source code:

![image](https://user-images.githubusercontent.com/10900667/95596726-232a3080-0a24-11eb-8317-21257b4e0b0f.png)

HttpRequest is a abstract class, how would you use that class without mocking?

**Reply by Tratcher** [reply]
Use `new DefaultHttpContext().Request`

**Reply by lommez** [reply]
Thank you so much @Tratcher !
It works perfectly!
